mla/analysis.py

```diff
@@ -153,11 +338,14 @@ def minimize_ts(self, events: np.ndarray, event_model: models.EventModel, # Supe
             test_ns = n_events - 0.00001
 
         drop = n_events - np.sum(preprocessing[1] != 0)
+        drop_index = preprocessing[1] != 0
+        preprocessing = (preprocessing[0][drop_index],preprocessing[1][drop_index]) # Ask Jason about this
 
```

Comment:
explain? This runs preprocessing twice.

Reply:
This part is actually handling dropping the events having zero signal spatial. I think the old implementation actually double count those events in the likelihood.